src/engine/RuntimeInformation.cpp

```diff
@@ -191,14 +197,14 @@ void RuntimeInformation::addLimitOffsetRow(const LimitOffsetClause& l,
   if (!(hasLimit || hasOffset)) {
     return;
   }
-  children_ = std::vector{*this};
-  auto& actualOperation = children_.at(0);
-  numRows_ = l.actualSize(actualOperation.numRows_);
+  children_ = std::vector{std::make_shared<RuntimeInformation>(*this)};
+  const auto& actualOperation = children_.at(0);
```

Comment:
It would be somewhat desirable to get only a shallow copy (i.e. to store the `shared_ptr` that already points to `this` (to which we of course don't have access without explicitly passing it in here).

Reply:
(Can we use `enable_shared_from_this` for such use cases?)

Reply:
This already is a shallow copy in a sense. The `RuntimeInformation` represented by `this` is shallowly copied for the newly shared pointer. Then `this` is modified so its only child is the newly created copy, and the number of rows adjusted.
So the only redundancy in this operation is the copy of members that are modified afterwards. In this case this means there are 3 non trivial copies that are "unecessary": one redundant `std::string` copy for `descriptor_`, one redundant copy of `std::vector` for `children_` and one redundant copy of `nlohmann::basic_json` for `details_`.

Using `enable_shared_from_this` would not help at all to solve this problem, because `children_ = std::vector{this->shared_from_this()}` would simply introduce a circular shared pointer reference (preventing it from ever deallocating itself, because itself holds a reference to itself inside the `children_` vector) and just drop any children that were there before. It just doesn't work this way, there needs to be a copy at some point.

However what could be done instead (theoretically at least, it doesn't make the code any easier to read) it to give `RuntimeInformation` another constructor that allows to copy the values that need to be copied anyways and move constructs from the 3 fields I mentioned earlier to avoid the copies. But that architecture really smells, thus I wouldn't recommend it.

What needs to be done conceptually (in my opinion at least), is to make all the operations "limit aware" in a sense so that the `RuntimeInformation` doesn't need to be modified afterwards, but it becomes an attribute that is propagated down to child operations depending on the type of operation (Filter could stop filtering after reaching the limit for example). Then this member function would be completely redundant